src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -802,33 +803,57 @@ pub fn add_ssts_to_sub_level(
         }
         return;
     }
-    add_new_sub_level(l0, insert_sub_level_id, level_type, insert_table_infos);
+    insert_new_sub_level(
+        l0,
+        insert_sub_level_id,
+        level_type,
+        insert_table_infos,
+        false,
+    );
 }
 
-pub fn add_new_sub_level(
+pub fn insert_new_sub_level(
     l0: &mut OverlappingLevel,
     insert_sub_level_id: u64,
     level_type: LevelType,
     insert_table_infos: Vec<SstableInfo>,
+    is_append: bool,
```

Comment:
I think instead of passing a `is_append` flag, we can pass a `insert_idx` to represent the position to insert, and `insert_idx == sub_levels.len()` means `is_append == true`. 

And we don't have to do binary search here, because in `init_with_parent_group`, we should already know the position index to insert when we find that the `sub_level_id` gets greater than `target_sub_level_id` while we iterate the sub level list from front to back.

Reply:
yes, I think it would be better to explicitly let the caller pass pos so that the caller maintains the logic for calculating pos and can clearly distinguish whether it is append or insert.
Wrong pos will bring data errors, so we need a strict sanity check to ensure the order of sub_level_id, after insert

Reply:
> yes, I think it would be better to explicitly let the caller pass pos so that the caller maintains the logic for calculating `pos` and can clearly distinguish whether it is `append` or `insert`. Wrong `pos` will bring data errors, so we need a strict sanity check to ensure the order of `sub_level_id`, after `insert`

ok